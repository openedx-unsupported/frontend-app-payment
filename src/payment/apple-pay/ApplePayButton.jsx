import React from 'react';
import PropTypes from 'prop-types';
import { performApplePayPayment } from './service';


export default class ApplePayButton extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      canMakePayments: global.ApplePaySession && global.ApplePaySession.canMakePayments(),
    };
  }

  handleClick = () => {
    performApplePayPayment({
      totalAmount: this.props.totalAmount,
      onPaymentBegin: this.props.onPaymentBegin,
      onPaymentComplete: this.props.onPaymentComplete,
      onMerchantValidationFailure: this.props.onMerchantValidationFailure,
      onPaymentAuthorizationFailure: this.props.onPaymentAuthorizationFailure,
      onPaymentCancel: this.props.onPaymentCancel,
    });
  }

  render() {
    if (!this.state.canMakePayments || this.props.totalAmount === undefined) return null;

    const other = { ...this.props };

    delete other.totalAmount;
    delete other.onPaymentBegin;
    delete other.onPaymentComplete;
    delete other.onMerchantValidationFailure;
    delete other.onPaymentAuthorizationFailure;
    delete other.onPaymentCancel;

    return (
      <button
        {...other}
        id="applePayBtn"
        onClick={this.handleClick}
        className="apple-pay-button"
        lang={this.props.lang}
        title={this.props.title}
      />
    );
  }
}

ApplePayButton.propTypes = {
  totalAmount: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onPaymentBegin: PropTypes.func,
  onPaymentComplete: PropTypes.func,
  onMerchantValidationFailure: PropTypes.func,
  onPaymentAuthorizationFailure: PropTypes.func,
  onPaymentCancel: PropTypes.func,
  lang: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

ApplePayButton.defaultProps = {
  totalAmount: undefined,
  onPaymentBegin: undefined,
  onPaymentComplete: undefined,
  onMerchantValidationFailure: undefined,
  onPaymentAuthorizationFailure: undefined,
  onPaymentCancel: undefined,
};
