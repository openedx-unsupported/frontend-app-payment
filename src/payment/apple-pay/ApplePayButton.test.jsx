/* eslint-disable global-require */
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import ApplePayButton from './ApplePayButton';


const applePaySession = { begin: jest.fn() };
global.ApplePaySession = jest.fn().mockImplementation(() => applePaySession);
global.ApplePaySession.canMakePayments = () => true;


describe('<ApplePayButton />', () => {
  it('should render properly', () => {
    const tree = renderer
      .create((
        <ApplePayButton
          totalAmount={10}
          lang="en"
          title="Pay with Apple Pay"
        />
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should perform an apple pay transaction on click', () => {
    const wrapper = mount((
      <ApplePayButton
        totalAmount={10}
        lang="en"
        title="Pay with Apple Pay"
      />
    ));

    wrapper.find('button').simulate('click');
    expect(applePaySession.begin).toHaveBeenCalled();
  });
});
