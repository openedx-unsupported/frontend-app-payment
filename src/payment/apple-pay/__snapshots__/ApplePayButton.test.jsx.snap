// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplePayButton /> should render properly 1`] = `
<button
  className="apple-pay-button"
  id="applePayBtn"
  lang="en"
  onClick={[Function]}
  title="Pay with Apple Pay"
/>
`;
